wazuh/wazuh-dashboard-plugins
# Allow edit groups from an agent

### Description
In order to allow users to edit the groups from an agent, we have to introduce the following changes:
- Modify the agent's table actions to add the option `Edit groups`.
- Add a new generic component `WzElementPermissions` to manage permission when we need a component different to a button.
- Migrate the `AgentsTable` class component to a functional component because we need to use an existing hook to disable the `Edit groups` action button.
 
### Issues Resolved
https://github.com/wazuh/wazuh-dashboard-plugins/issues/6138

### Evidence
#### Actions menú
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/103193307/536bdf4a-571f-4a98-9776-ca6f807113fa)

#### Actions menú when never connected
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/103193307/fbdaa8c5-e482-4281-a735-5229d4b592ff)
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/103193307/b5ff9cb2-e8a9-439d-837d-d235ce0fa8f3)

#### `Edit groups` modal
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/103193307/4038e296-3780-4cde-a75f-3b8733b3dd1c)

#### Actions menú with read only user
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/103193307/e3409479-af0b-4518-87cd-f9123772fefd)
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/103193307/88fd0fcb-4034-40a4-992f-1cc5e270c42b)

#### `Deploy new agent` button with readonly user
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/103193307/3fd009de-2233-4556-af2a-23ebcf524729)



### Test
- Go to Endpoints Summary
- Verify that actions work consistently with agent state and user permissions.
- Test buttons that have permission validations
- Test table filters
To create a readonly user: https://documentation.wazuh.com/current/user-manual/user-administration/rbac.html#creating-and-setting-a-wazuh-read-only-user

### Check List
